src/app/package/page.tsx

```diff
@@ -46,7 +55,14 @@ export default function PackageList() {
     }
   }, [data])
 
-  return (
+  return isLoadingFreePackage ||
+    isLoadingPremiumPackage ||
+    isLoadingLatestSubscription ||
+    !free_package ||
+    !premium_package ||
+    !latest_subscription ? (
```

Comment:
This is a bit too much condition isn't it ? Is there a reaosn why you can't simplyfy it with just the isLoading only ?

Reply:
As you can see, I have to wait for each fetch from the backend to complete. Each variable represents a separate fetch operation, hence the need for multiple conditions. This ensures that all necessary data—such as the latest subscription, free package, and premium package—are fully loaded before the screen displays. Simplifying it to just the `isLoading` condition would not account for the individual completion of each data fetch.

Reply:
I agree with Natania's answer, I think it's fine to use everything because it fetches from a lot of different query (also different endpoints), their load time couldn't be unified into 1 variable